crates/precompile/src/bls12_381/g1_msm.rs

```diff
@@ -54,8 +49,11 @@ pub(super) fn g1_msm(input: &Bytes, gas_limit: u64) -> PrecompileResult {
         if encoded_g1_element.iter().all(|i| *i == 0) {
             continue;
         }
+
+        let [a_x, a_y] = remove_g1_padding(encoded_g1_element)?;
```

Comment:
removing the padding is blst agnostic, so it has been moved into utils.rs from blst.rs